doc/book/cookbook.md

```diff
@@ -228,7 +228,55 @@ return [
                     'ValidationMiddleware',
                 ],
             ],
-        ], 
+        ],
     ],
 ];
 ```
+
+## How can I set custom 404 page handling?
+
+We can set custom 404 page handling instead of [using the final handler](http://zend-expressive.readthedocs.org/en/latest/error-handling/) for logging or other purpose with create new middleware that we want to intercept if no other middleware has executed, and is indicate a "not found" situation.
```

Comment:
@weierophinney I'm not quite sure about final handler part, would you mind to tackle for it ? Thank you ^^